### PR TITLE
841 fe migrate counter to UI

### LINKED ISSUE #841: [FE] Migrate Counter to UI

### PR DESCRIPTION
**Pull request by marmorrei**
Closes #841

### CODE REVIEW

File: packages/ui/src/components/atoms/Spinner.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  big: '133px',
+}
+
+type TSpinner = {
+  size?: 'xsmall' | 'small' | 'medium' | 'big'
+}
+
+const Spinner = styled.div.attrs<TSpinner>((props) => ({
+  size: props?.size,
+}))<TSpinner>`
```
Comment (NataliaR-BCN) [inline packages/ui/src/components/atoms/Spinner.ts:26]: Is there any console warning with the prop?

File: packages/ui/src/stories/atoms/Spinner.stories.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    },
+  },
+} satisfies Meta<typeof Spinner>
+
+export default meta
+type SpinnerStory = StoryObj<typeof meta>
+
+export const Default: SpinnerStory = {
+  args: {
+    size: 'xsmall',
```
[the remark this answers is not in the document]
Comment (marmorrei) [reply packages/ui/src/stories/atoms/Spinner.stories.ts:25]: It is for the Counter component as far as I know. Should I just name it ExtraSmall and not add a Default?

File: packages/ui/src/components/atoms/Spinner.ts (hunk 1 above)
Comment (marmorrei) [reply packages/ui/src/components/atoms/Spinner.ts:26]: Nothing about props.

File: packages/ui/src/stories/atoms/Spinner.stories.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (NataliaR-BCN) [reply packages/ui/src/stories/atoms/Spinner.stories.ts:25]: I mean: size is an optional prop in Spinner, maybe the default version should be without any size.

[the remark this answers is not in the document]
Comment (marmorrei) [reply packages/ui/src/stories/atoms/Spinner.stories.ts:25]: I will try that.

File: packages/ui/src/__tests__/atoms/Spinner.test.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      'border-top-color': `${colors.primary}`,
+    })
+    expect(screen.getByTestId('spinner')).toHaveStyle({
+      'border-right-color': `${colors.primary}`,
+    })
+    expect(screen.getByTestId('spinner')).toHaveStyle({
+      'border-radius': '50%',
+    })
+  })
+})
```
[the remark this answers is not in the document]
Comment (marmorrei) [reply packages/ui/src/__tests__/atoms/Spinner.test.tsx:136]: I have added tests for default and for each size.

File: packages/ui/src/stories/atoms/Spinner.stories.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (marmorrei) [reply packages/ui/src/stories/atoms/Spinner.stories.ts:27]: It is done. I didn't check well the default size. It is medium. All changed.

File: packages/ui/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,11 @@
 # Changelog
 
 All notable changes to this project will be documented in this file.
+## [0.11.0] - 2024-01-24
```
Comment (kevinmamaqi) [inline packages/ui/CHANGELOG.md:4]:
```suggestion
## [0.11.0] - 2024-01-24

```

File: packages/ui/src/__tests__/atoms/Spinner.test.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
+import { render, screen } from '@testing-library/react'
+import { Spinner } from '../../components/atoms/index'
+import { colors } from '../../styles'
+
+describe('Spinner', () => {
+  it('renders correctly by default', () => {
+    render(<Spinner data-testid="spinner" />)
+
+    // defaultSize === medium
+    const defaultSize = 70
```
Comment (kevinmamaqi) [inline packages/ui/src/__tests__/atoms/Spinner.test.tsx:10]: borrar comentario y declarar fuera del test.

File: packages/ui/src/__tests__/atoms/Spinner.test.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    })
+    expect(screen.getByTestId('spinner')).toHaveStyle({
+      'border-radius': '50%',
+    })
+  })
+
+  it('renders correctly big', () => {
+    render(<Spinner size="big" data-testid="spinner" />)
+
+    const big = 133
```
Comment (kevinmamaqi) [inline packages/ui/src/__tests__/atoms/Spinner.test.tsx:114]: Definir todos los sizes fuera

File: packages/ui/src/__tests__/atoms/Spinner.test.tsx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+    expect(screen.getByTestId('spinner')).toHaveStyle({
+      border: `${big / 6}px solid ${colors.outlineHover}`,
+    })
+    expect(screen.getByTestId('spinner')).toHaveStyle({
+      'border-top-color': `${colors.primary}`,
+    })
+    expect(screen.getByTestId('spinner')).toHaveStyle({
+      'border-right-color': `${colors.primary}`,
+    })
+    expect(screen.getByTestId('spinner')).toHaveStyle({
```
Comment (kevinmamaqi) [inline packages/ui/src/__tests__/atoms/Spinner.test.tsx:132]: Excesivo testear el border-radius, border, color y demás en cada test. Lo unico que cambia es el prop size.

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

**Review by NataliaR-BCN — DISMISSED** [review]

**Review by NataliaR-BCN — APPROVED** [review]

**Review by REALcuba — APPROVED** [review]

**Review by kevinmamaqi — APPROVED** [review]

**Pull request merged**